notification.go

```diff
@@ -0,0 +1,78 @@
+package watchdog
+
+import "sync"
+
+var (
+	gcNotifeeMutex sync.Mutex
+	gcNotifees     []notifeeEntry
+
+	forcedGCNotifeeMutex sync.Mutex
+	forcedGCNotifees     []notifeeEntry
+)
+
+// RegisterNotifee registers a function that is called when a GC has happened.
+// The unregister function returned can be used to unregister this notifee.
+func RegisterNotifee(f func()) (unregister func()) {
+	gcNotifeeMutex.Lock()
+	defer gcNotifeeMutex.Unlock()
+
+	var id int
+	if len(gcNotifees) > 0 {
+		id = gcNotifees[len(gcNotifees)-1].id + 1
+	}
+	gcNotifees = append(gcNotifees, notifeeEntry{id: id, f: f})
+
+	return func() {
+		gcNotifeeMutex.Lock()
+		defer gcNotifeeMutex.Unlock()
+
+		for i, entry := range gcNotifees {
+			if entry.id == id {
+				gcNotifees = append(gcNotifees[:i], gcNotifees[i+1:]...)
+			}
+		}
+	}
+}
+
+func notifyGC() {
+	if NotifyGC != nil {
+		NotifyGC()
+	}
+	gcNotifeeMutex.Lock()
+	defer gcNotifeeMutex.Unlock()
+	for _, entry := range gcNotifees {
+		entry.f()
+	}
+}
+
+// RegisterForcedGCNotifee registers a function that is called before watchdog triggers a GC run.
+// The unregister function returned can be used to unregister this notifee.
+func RegisterForcedGCNotifee(f func()) (unregister func()) {
```

Comment:
Let's call this `RegisterPreGCNotifee`, and specify that this will be called _only_ on watchdog initiated GC in Godocs.

Reply:
Done.